src/nvim/globals.h

```diff
@@ -727,6 +727,7 @@ EXTERN bool listcmd_busy INIT(= false);     // set when :argdo, :windo or
                                             // :bufdo is executing
 EXTERN bool need_start_insertmode INIT(= false);
 // start insert mode soon
+EXTERN char *last_mode INIT(= NULL);
```

Comment:
Diverges from the patch.
I don't mind because of the NULL check in trigger_modechanged().

Reply:
This diverges because I am using an array in vim and a pointer in the port here. I use a pointer here, because `get_mode()` returns allocated memory. I agree, that this is fine because of the NULL check.

Reply:
BTW the null check is removed again in the port of 8.2.3555, because `last_mode` cannot ever be `NULL` anymore, as it is now initialized in `autocmd.c` when setting up the first `ModeChanged` autocmd.